src/index.js

```diff
@@ -1,22 +1,22 @@
 // @flow
-type EqualityFn = (a: any, b: any) => boolean;
+type EqualityFn = (a: mixed, b: mixed) => boolean;
 
-const simpleIsEqual = (a: any, b: any): boolean => a === b;
+const simpleIsEqual: EqualityFn = (a: mixed, b: mixed): boolean => a === b;
 
-export default function (resultFn: Function, isEqual?: EqualityFn = simpleIsEqual) {
-  let lastThis: any;
-  let lastArgs: Array<any> = [];
-  let lastResult: any;
+export default function <ResultFn: (...Array<any>) => mixed>(resultFn: ResultFn, isEqual?: EqualityFn = simpleIsEqual): ResultFn {
```

Comment:
@thejameskyle `<ResultFn: (...Array<any>) => mixed>` seemed to work fine. I am still a little confused as to what this is communicating. Here is my current understanding:

`ResultFn` is a type of function (as communicated by the syntax`() => {}`) that can accept any length of arguments (`(...Array<any>)`) and returns a `mixed` (which means that you must check the result type before you can do type specific operations on it).

Is that correct?